worker/src/features/batchExport/handleBatchExportJob.ts

```diff
@@ -16,12 +19,193 @@ import {
   sendBatchExportSuccessEmail,
   streamTransformations,
   BatchExportJobType,
+  createTracesQuery,
+  orderByToPrismaSql,
+  tableColumnsToSqlFilterAndPrefix,
 } from "@langfuse/shared/src/server";
 
 import { env } from "../../env";
 import { logger } from "@langfuse/shared/src/server";
+import { BatchExportSessionsRow, BatchExportTracesRow } from "./types";
+
+const tableNameToTimeFilterColumn = {
+  sessions: "created_at",
+  traces: "timestamp",
+};
+
+const isTimestampFilter = (filter: FilterCondition): filter is TimeFilter => {
+  return filter.column === "Timestamp" && filter.type === "datetime";
+};
+
+const parseTracesOrderByAndFilter = (
+  orderBy: BatchExportQueryType["orderBy"],
+  filter: BatchExportQueryType["filter"]
+) => {
+  const orderByCondition = orderByToPrismaSql(orderBy, tracesTableCols);
+  const filterCondition = tableColumnsToSqlFilterAndPrefix(
+    filter ?? [],
+    tracesTableCols,
+    "traces"
+  );
+
+  const scoreTimestampFilter = filter?.find(isTimestampFilter);
+
+  return {
+    orderByCondition,
+    filterCondition,
+    scoreTimestampFilterCondition: scoreTimestampFilter
+      ? Prisma.sql`AND s.timestamp >= ${scoreTimestampFilter.value}`
+      : Prisma.empty,
+  };
+};
+
+const getDatabaseReadStream = async ({
+  projectId,
+  tableName,
+  filter,
+  orderBy,
+  cutoffCreatedAt,
+}: {
+  projectId: string;
+  cutoffCreatedAt: Date;
+} & BatchExportQueryType): Promise<DatabaseReadStream<unknown>> => {
+  // Set createdAt cutoff to prevent exporting data that was created after the job was queued
+  const createdAtCutoffFilter: FilterCondition = {
+    column: tableNameToTimeFilterColumn[tableName],
+    operator: "<",
+    value: cutoffCreatedAt,
+    type: "datetime",
+  };
+
+  switch (tableName) {
+    case "sessions":
+      // TODO: improve performance for (count(*) OVER ())::int AS "totalCount"
```

Comment:
_See comment below_

Adding comment to resolve after removing todo before merging